### PR TITLE
Convert code to C# 6.0

### PR DESCRIPTION
**Pull request by Sharparam**
**Development will now require Visual Studio 2015 or higher**

Colore has been converted to use C# 6.0 features where possible, as well as migrating to the new `StyleCop.Analyzers` package (which supports C# 6.0 as well as doing some other changes).

**Static code analysis is now enabled on Release builds** (using the recommended ruleset for managed code from Microsoft).

`SecuritySafeCriticalAttribute` has been added to code doing P/Invoke or other forms of native code interop, **this should be verified to not be exploitable**.

### :warning: :bangbang: **NOT INTEGRATION TESTED** :bangbang: :warning:

The builds generated under C# 6.0 **must be integration tested** before merging this PR. The following table shows needed integration tests as of 2015-12-20:

Platform | Tested?    | Link to build artifact
-------- | ---------- | ----------------------
AnyCPU   | **NO** :x: | [colore_317_anycpu.zip][anycpu]
x86      | **NO** :x: | [colore_317_x86.zip][x86]
x64      | **NO** :x: | [colore_317_x64.zip][x64]

[anycpu]: http://tc.coralestudios.com/repository/download/colore_mainbuild/902:id/colore_317_anycpu.zip
[x64]: http://tc.coralestudios.com/repository/download/colore_mainbuild/902:id/colore_317_x64.zip
[x86]: http://tc.coralestudios.com/repository/download/colore_mainbuild/902:id/colore_317_x86.zip

### CODE REVIEW

File: Corale.Colore/stylecop.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
...ight © 2015 by Adam Hellberg and Brandon Scott.\n\n    Permission is hereby granted, free of charge, to any person obtaining a copy of\n    this software and associated documentation files (the \"Software\"), to deal in\n    the Software without restriction, including without limitation the rights to\n    use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies\n    of the Software, and to permit persons to whom the Software is furnished to do\n    so, subject to the following conditions:\n\n    The above copyright notice and this permission notice shall be included in all\n    copies or substantial portions of the Software.\n\n    THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\n    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\n    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\n    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,\n    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN\n    CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.\n\n    \"Razer\" is a trademark of Razer USA Ltd.",
```
Comment (brandonscott) [inline Corale.Colore/stylecop.json]: Any reason this has spaces at the start?

**Comment by brandonscott** [comment]
Really glad we're moving to C# 6.0.  I've been using it for a couple of months and I love the new features it brings.

Thanks @Sharparam for implementing these changes.

The Integration Testing Tool is being worked on in the background and I hope to have it finished by the New Year.  In the mean time, anyone that manually tests these changes, please let us know on Gitter or by email.

File: Corale.Colore/stylecop.json (hunk 1 above)
Comment (Sharparam) [inline Corale.Colore/stylecop.json]: It's to follow the style we had on the header copyright notice where it is indented inside the xml tag:

```
<copyright ...>
    Copyright text here
</copyright>
```

**Comment by Sharparam** [comment]
Or let us know with a comment on this PR to keep it in one place, also makes it easier for me to update the table in the main comment.

**Pull request merged**